docs/secure-communication-agents.asciidoc

```diff
@@ -1,21 +1,353 @@
-. <<https-in-agents,Enable HTTPS in your APM Agents>>
+As soon as secure communication is enabled, requests without a valid token or API key will be denied by APM Server.
+As RUM endpoints cannot be secured, they are exempt from this rule.
+
+* <<api-key,API keys>>
+* <<secret-token,Secret token>>
+
+[[api-key]]
+=== API keys
+
+// Most operations require the `manage_security` cluster privilege,
+// and you must ensure that `apm-server.api_key` or `output.elasticsearch` are configured appropriately.
+// If you modify or delete apm privileges, APM Server might reject all requests.
+// APM Server will create security privileges for the `apm` application; you can freely query them.
+// If you modify or delete apm privileges, APM Server might reject all requests.
+// Check the Elastic Security API documentation for details.
+
+You can configure an API key to authorize requests to the APM Server.
+API keys are sent as plain-text,
+so they only provide security when used in combination with SSL/TLS.
+When you set an API key in each of your agents,
+you ensure that only your agents are able to send data to your APM servers.
+
+To secure the communication between APM Agents and the APM Server with API keys:
+
+. <<ssl-setup-api,Enable SSL/TLS>>
+. <<configure-api-key,Enable and configure API keys>>
+. <<create-api-key,Create an API key with APM Server>>
+. <<set-api-key,Set the API key in your APM Agents>>
+
+NOTE: API Keys are not applicable for the RUM Agent,
+as there is no way to prevent them from being publicly exposed.
+
+[[ssl-setup-api]]
+[float]
+=== Enable SSL/TLS communication
+
+// Use the shared ssl short description
+include::./ssl-input-short.asciidoc[]
+
+[[configure-api-key]]
+[float]
+=== Enable and configure API keys
+
+API keys are disabled by default. You can change this, and additional settings,
+in the `apm-server.api_key` section of the +{beatname_lc}.yml+ configuration file.
+
+At a minimum, you must enable API keys,
+and should set a limit on the number of unique API keys {es} allows per minute.
+Here's an example `apm-server.api_key` config using 50 unique API keys:
+
+[source,yaml]
+----
+apm-server.api_key.enabled: true <1>
+apm-server.api_key.limit: 50 <2>
+----
+<1> Enables API keys
+<2> Restricts the number of unique API keys that {es} allows each minute.
+This value should be the number of unique API keys configured in your monitored services.
+
+All other configuration options are described in <<api-key-settings>>.
+
+[[create-api-key]]
+[float]
+=== Create and validate an API key
+
+APM Server provides a command line interface for creating API keys.
+Keys created using this method can only be used for Agent/Server communication.
+
+While it is possible to CURL {es} directly to create API keys,
+using this method is not recommended.
+The command line tool has safeguards that ensure the required privileges exist before creating an API key.
+If they do not exist, the CLI tool will create the privileges.
```

Comment:
Those two sentences are not valid anymore.

Reply:
Which two sentences? These?
```
The command line tool has safeguards that ensure the required privileges exist before creating an API key.
If they do not exist, the CLI tool will create the privileges.
```

Reply:
So does this mean there are now no differences between using the CLI tool vs CURLing ES? Other than, of course, the CLI tool makes it much easier to create/validate/delete keys.

Reply:
Clarified with Juan and removed.